rewindable-index/src/RewindableIndex/Storable.hs

```diff
@@ -189,16 +193,31 @@ newtype Config = Config
   } deriving (Show, Eq)
 $(Lens.makeLenses ''Config)
 
+data SyntheticEvent e p =
```

Comment:
Why not `data SyntheticEvent h = Event !(StorableEvent h) | Synthetic !(StorablePoint h)`?

Reply:
It complicates reasoning about the type-system when you need to specify further constraints in the `e` and `p` type variables, leading to the need to enable several ghc extensions. I don't see what exactly we would gain from that.

Reply:
:+1: